Add station names to codes mapping

shriakhilc opened the pull request:
Refers #17 

Station name to code mapping added as per an official Indian Railways document [here](http://www.indianrailways.gov.in/railwayboard/uploads/directorate/coaching/pdf/Station_code.pdf). It contains **711 stations** total.

While other sources list up to _8000_ stations, those also seem to include goods stations where passenger trains don't usually stop. Nonetheless, I will still be scraping them and updating the file in the future.

shriakhilc committed:
Add station names to codes table

Add a SQLite database with a table `station_codes`

Refers #17

prinzpiuz commented:
anyway great attempt but
no need of hard coding things try to implement a inline command to auto suggest station code while we typing

prinzpiuz commented:
sorry for the comment . i didnt noticed it actually will test and tell  you

prinzpiuz commented:
but making such large dict/json will  effect perfomance issue...?

shriakhilc commented:
There are 3 common persistence solutions in Python that can be used - The `pickle` module, the `sqlite3` module and using external files, commonly JSON, YAML, or XML.

I browsed a number of posts and checked the performance statistics over the years. This [StackOverflow Answer](https://stackoverflow.com/a/47764585/6698642) is the latest one for that question regarding pickle vs json performance. It seems that json used to be faster for `str` initially, but `pickle` should be faster now in most cases. Note that the latest Python3 `pickle` module internally uses the faster `cPickle` implementation if possible, and `cPickle` is no longer available as a separate module.  

Now, whether to use `pickle` or `sqlite3` depends on what will be done with the data. Pickle simply stores and retrieves the entire `dict` object as a byte stream, whereas sqlite3 can provide proper query support. 

If whatever "fuzzy search" that is planned to be implemented can be done with a simple SQL Query, then `sqlite3` would be a better choice. Otherwise, the entire `dict` can be loaded as an object, and can be used as the source for said search operation (whether from a library, or implemented yourselves).

I can update the PR with a suitable file depending on your choice.

prinzpiuz commented:
i think sqlite3 will be the better choice otherwise if entire dict loaded as object it will effect the perfomance i think .....also can you join this group for more discussions with other members... https://t.me/RailgadiBotDiscussions

shriakhilc commented:
Updated the PR. Switched from the JSON file to using a SQLite database.

The name of the database is `railwaybot.db` and it has a table called `station_codes` with two columns `name` and `code` both of type `TEXT`. 

Documentation for using Python 3 's sqlite3 module : [sqlite3](https://docs.python.org/3/library/sqlite3.html)

The added advantage of using a SQLite database is that we will be able to filter the station names based on current user input using wildcard characters and the `WHERE` and `LIKE` keywords. 

For example if the user is typing "Lucknow" :
```
select * from station_codes where name like "luck%"
```

This will be useful when there are many stations within the same city, as with a lot of larger ones.

shriakhilc commented:
@prinzpiuz can this be merged now?

prinzpiuz commented:
need to be tested and need to make bot inline also then only i think this feature works... everybody in group is busy i think

prinzpiuz commented:
let me make inline commands
